pkg/controllers/repositoryserver/repository.go

```diff
@@ -52,3 +57,20 @@ func (h *RepoServerHandler) connectToKopiaRepository() error {
 		args,
 	)
 }
+
+func (h *RepoServerHandler) getRepositoryCacheSettings() (contentCacheMB, metadataCacheMB int, err error) {
+	contentCacheMB, metadataCacheMB = command.GetGeneralCacheSizeSettings()
+	if h.RepositoryServer.Spec.Repository.CacheSizeSettings.Content != "" {
```

Comment:
Can `CacheSizeSettings` also be nil here and needs a nil check?

Reply:
its not a pointer, hence it cannot be nil. it will always have `{}` value when its unmarshalled even if its not present in the CRD.